src/Stratis.Features.SQLiteWalletRepository/SQLiteWalletRepository.cs

```diff
@@ -754,9 +757,12 @@ public void ProcessBlocks(IEnumerable<(ChainedHeader header, Block block)> block
 
                     foreach ((ChainedHeader header, Block block) in blocks.Append((null, null)))
                     {
+                        bool done = false;
+
                         try
                         {
-                            ParallelProcessBlock(round, block, header);
+                            if (!ParallelProcessBlock(round, block, header))
```

Comment:
This type of logic seems counter-intuitive. I would think that if Process Blocks returns false we should do something else, not break? I..e if we are still processing return false and if we are done return true and break?

Also, this `foreach` and the one on line 722 are identical so perhaps refactor to a method so that we avoid possible issues in future.

Reply:
`ParallelProcessBlock` returns `false` if the necessary wallet locks can't be acquired to continue processing. In this case we simply want to exit and wait for the next scheduled synchronization to occur. It also returns `false` if it can't continue due to running out of blocks to process.